galley/pkg/config/analysis/diag/messages.go

```diff
@@ -14,10 +14,38 @@
 
 package diag
 
+import "sort"
+
 // Messages is a slice of Message items.
 type Messages []Message
 
 // Add a new message to the messages
 func (ms *Messages) Add(m Message) {
 	*ms = append(*ms, m)
 }
+
+// Sort the message lexicographically by level, code, origin, then string.
+func (ms Messages) Sort() {
+	sort.Slice(ms, func(i, j int) bool {
+		switch {
+		case ms[i].Type.Level() != ms[j].Type.Level():
+			return ms[i].Type.Level() < ms[j].Type.Level()
```

Comment:
The sort on Level should probably be by severity, not the string value. Error -> Info -> Warning is a weird order to read things in.